### PR TITLE
Fix BoxView not rendering when is hidden on start

### PR DESCRIPTION
**Pull request by kubaflo**
### Description of Change

I observed that the BoxView did appear on the screen when the user changed the property `IsVisible` from false to true when `IsVisible = false` was set on the start.

This is a hotfix to this PR: https://github.com/dotnet/maui/pull/21322

|Before|After|
|--|--|
|<video src="https://github.com/dotnet/maui/assets/42434498/beea1639-05c6-40fe-bbf0-0242ec4a7028" width="300px"/>|<video src="https://github.com/dotnet/maui/assets/42434498/78a8cd24-0115-4634-8b61-96f03aade5ee" width="300px"/>|

### CODE REVIEW

File: src/Controls/src/Core/BoxView/BoxView.cs [hunk 1]
```diff
@@ -56,6 +56,7 @@ protected override void OnPropertyChanged([CallerMemberName] string? propertyNam
 
 			if (propertyName == BackgroundColorProperty.PropertyName ||
 				propertyName == ColorProperty.PropertyName ||
+				propertyName == IsVisibleProperty.PropertyName ||
```
Comment (jsuarezruiz) [inline src/Controls/src/Core/BoxView/BoxView.cs:59]: Could we include a related UITest?

File: src/Controls/src/Core/BoxView/BoxView.cs [hunk 2]
```diff
@@ -56,9 +56,13 @@ protected override void OnPropertyChanged([CallerMemberName] string? propertyNam
...
 				propertyName == BackgroundProperty.PropertyName ||
 				propertyName == CornerRadiusProperty.PropertyName)
 				Handler?.UpdateValue(nameof(IShapeView.Shape));
+
+			if(propertyName == IsVisibleProperty.PropertyName)
```
Comment (jsuarezruiz) [inline src/Controls/src/Core/BoxView/BoxView.cs:64]: Is this necessary?

Comment (kubaflo) [reply src/Controls/src/Core/BoxView/BoxView.cs:64]: Strangely enough, it is only necessary for Android. For iOS everything works with only updating `Handler?.UpdateValue(nameof(IShapeView.Shape));`  when the visibility changes

File: src/Controls/tests/UITests/Tests/Issues/Issue19926.cs [hunk 3]
```diff
@@ -16,7 +16,10 @@ public Issue19926(TestDevice device)
 	public void PropertiesShouldBeCorrectlyApplied()
 	{
 		this.IgnoreIfPlatforms(new TestDevice[] { TestDevice.iOS, TestDevice.Mac, TestDevice.Windows });
+
 		_ = App.WaitForElement("boxView");
+		App.Click("button");
+		_ = App.WaitForElement("boxView2");
```
Comment (jsuarezruiz) [inline src/Controls/tests/UITests/Tests/Issues/Issue19926.cs:22]: Include a small delay here before take the snapshot to wait button ripple effect animation to complete.

Comment (kubaflo) [reply src/Controls/tests/UITests/Tests/Issues/Issue19926.cs:22]: Done. I hope 500ms is enough

**Comment by jsuarezruiz** [comment]
@kubaflo Could you rebase?. I think this PR potentially could also fix https://github.com/dotnet/maui/issues/22897

**Review by jsuarezruiz — DISMISSED** [review]
The test **PropertiesShouldBeCorrectlyApplied** is failing on Android with the changes:
image

**Comment by kubaflo** [comment]
@jsuarezruiz could you /azp pls?

**Review by PureWeen — APPROVED** [review]

**Pull request merged**